web/frontend/src/hooks/robot-client.ts

```diff
@@ -25,28 +25,42 @@ const context = {
 
 export const useRobotClient = () => context;
 
+export type DisconnectCallback = () => void;
+export type ConnectCallback = (() => DisconnectCallback) | (() => void);
+
 /**
- * This hook will fire whenever a connection occurs.
+ * Pass a callback to this hook that will fire whenever an initial connection or reconnect occurs.
+ * 
+ * The callback can return a disconnection callback that will fire whenever a disconnect or unmount occurs.
+ * 
+ * @example
+ * ```ts
+ * useConnect(() => {
+ *   // This will fire after component mount, once a robot has connected, and on reconnect.
+ *   return () => {
+ *     // This will fire if a robot disconnects, or when the component is destroyed.
+ *   }
+ * })
+ * ```
  */
-export const useConnect = (callback: () => void) => {
+export const useConnect = (callback: ConnectCallback) => {
   const { connectionStatus } = useRobotClient();
 
-  const unsub = connectionStatus.subscribe((value) => {
-    if (value === 'connected') {
-      callback();
-    }
-  });
+  // eslint-disable-next-line @typescript-eslint/no-invalid-void-type
+  let disconnectCallback: DisconnectCallback | void
```

Comment:
```suggestion
  let disconnectCallback: DisconnectCallback | undefined
```

Reply:
Tried this initially, but TS in this case wants an explicit undefined 👿